File: javascript/horoscope-daily.mjs
// Daily horoscope for all zodiac signs

const API_KEY = process.env.ASTROAPI_KEY;
const BASE_URL = process.env.ASTROAPI_BASE_URL || "https://api.astroapi.cloud";

if (!API_KEY) {
    console.error("Error: Set ASTROAPI_KEY environment variable first.");
    process.exit(1);
}

const today = new Date().toISOString().split("T")[0];

const response = await fetch(`${BASE_URL}/api/horoscope/daily/${today}?language=en`, {
    headers: {
        "X-Api-Key": API_KEY
    }
});

if (response.status === 403) {
    console.error("Access denied: This endpoint requires the \"daily-report:read\" module.");
    console.error("Please ensure your API key has access to this module.");
    process.exit(1);
}

if (!response.ok) {
    console.error(`Error: ${response.status} ${response.statusText}`);
    console.error(await response.text());
    process.exit(1);
}

const data = await response.json();
console.log(JSON.stringify(data, null, 2));

File: python/horoscope_daily.py
# AstroAPI — Daily horoscope for today

import json
import os
import sys
from datetime import date

import requests

API_KEY = os.environ.get("ASTROAPI_KEY")
BASE_URL = os.environ.get("ASTROAPI_BASE_URL", "https://api.astroapi.cloud")

if not API_KEY:
    print("Error: Set ASTROAPI_KEY environment variable first.", file=sys.stderr)
    print('  export ASTROAPI_KEY="your-api-key"', file=sys.stderr)
    sys.exit(1)

HEADERS = {
    "X-Api-Key": API_KEY,
    "Content-Type": "application/json"
}

today = date.today().isoformat()

response = requests.get(
    f"{BASE_URL}/api/horoscope/daily/{today}",
    headers=HEADERS,
    params={"language": "en"}
)

if response.status_code == 403:
    print("Access denied: the 'daily-report:read' module is required for this endpoint.")
    print("Make sure your API key has this module enabled.")
    sys.exit(0)

response.raise_for_status()

data = response.json()
print(json.dumps(data, indent=2))
